### PR TITLE
Part of introductory doc

### PR DESCRIPTION
**Pull request by prybicki**

### CODE REVIEW

File: docs/design-doc.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+## Abstract
+
+THC is a toolset for particle simulations comprising of simulator, visualizer and analyzer.
+
+The simulator is an application which is capable of reading simulation script in form of YAML file, performing it and saving whole or part of simulation trajectory (properties of particles changing over time).
+
+The visualizer is a graphical program used to display saved simulations. It allows to watch it at adjustable pace, pan and zoom. 
+
+The analyzer is a tool for post-sipmulation computations of its statistics and meta-data, e.g. checking energy leak.
```
Comment (iwonapytel) [inline docs/design-doc.md]: sipmulation

File: docs/design-doc.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+## Simulator
+
+Simulator's implementation consist of following three parts: interpreter, data manager and a number of computational modules (kernels).
+
+The interpreter is able to verify simulation script correctness and perform it using computational modules and data manger.
```
Comment (iwonapytel) [inline docs/design-doc.md]: data manger

File: docs/design-doc.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+The data manager executes part of the simulation script that describe what data is used in the simulation and handles runtime requests for particle creation/destruction raised by computational kernels.
```
Comment (iwonapytel) [inline docs/design-doc.md]: that describes

File: docs/design-doc.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+Simulation data consists of any number of:
+- global variables
+- object groups
+
+Global variable stores single value of its type. Its type is unchangeable through simulation execution. A good examples of global variables are: realTimeElapsed, stepNumber, gravityConstant, etc.
+
+An object group is a storage for large number of objects utilizing structure-of-arrays model. 
+
+An object group consist of any number of:
```
Comment (iwonapytel) [inline docs/design-doc.md]: consists

File: docs/design-doc.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+- properties (arrays)
+- groupVariables
+
+Property is a continguous memory that store some attribute of all objects from the group. Note that all properties of the object group are of the same element count.
+
+The purpose of group variable is to save space and memory bandwith whenever it is known ahead-of-time that some attribute will have the same value for all objects in a group. It store a single value of its type and it is associated with its object group. 
+For example spring bonds between particles may share maximum length before breaking which is a good candidate for group variable.
+
+Declaration of either variable or property must specify its type, name and can specify initial value. In case of property, both type and initial value are specified per element.
```
Comment (iwonapytel) [inline docs/design-doc.md]: and can specify = and optional

File: docs/design-doc.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+All those rules are ilustrated in following scheme.
```
Comment (iwonapytel) [inline docs/design-doc.md]: illustrated

File: docs/design-doc.md (hunk 5 above)
Comment (prybicki) [reply docs/design-doc.md]: Must specify XYZ and optional ZYX?

**Pull request merged**